agent/taskresource/cgroup/cgroup.go

```diff
@@ -97,12 +109,95 @@ func (c *CgroupResource) GetName() string {
 	return resourceName
 }
 
+// DesiredTerminal returns true if the cgroup's desired status is REMOVED
+func (c *CgroupResource) DesiredTerminal() bool {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+
+	return c.GetDesiredStatus() == taskresource.ResourceStatus(CgroupRemoved)
```

Comment:
I think you can just compare `c.desiredStatusUnsafe == CgroupRemoved` here? same below

Reply:
Then you are acquiring the lock twice, which may cause deadlock.

Reply:
will change this to `c.desiredStatusUnsafe`